scheme/deltas/041-general-review.sql

```diff
@@ -0,0 +1,27 @@
+DROP INDEX peerServerId ON LcrGateways;
+DROP INDEX name_domain_idx ON Terminals;
+DROP INDEX terminalId ON Users;
+
+ALTER TABLE Users CHANGE active active TINYINT(1) UNSIGNED DEFAULT '0' NOT NULL; 
+ALTER TABLE Users CHANGE maxCalls maxCalls SMALLINT UNSIGNED DEFAULT 2 NOT NULL;
+
+/* Mysql's unique keys (`domain_users`) can not be larger than 255 characters: 191 domain char + 64 username char  */
+ALTER TABLE Companies CHANGE domain_users domain_users VARCHAR(191) NOT NULL;
+ALTER TABLE kam_users_domain_attrs CHANGE did did VARCHAR(255) DEFAULT NULL;
+ALTER TABLE kam_users_location_attrs CHANGE domain domain VARCHAR(191) DEFAULT NULL;
+ALTER TABLE kam_users_location CHANGE domain domain VARCHAR(191) DEFAULT NULL;
+
+ALTER TABLE kam_users_xcap CHANGE domain domain VARCHAR(191) NOT NULL;
+ALTER TABLE kam_users_active_watchers CHANGE to_domain to_domain VARCHAR(191) NOT NULL, CHANGE from_domain from_domain VARCHAR(191) NOT NULL;
+ALTER TABLE kam_users_watchers CHANGE watcher_domain watcher_domain VARCHAR(191) NOT NULL;
+ALTER TABLE kam_trunks_uacreg CHANGE l_domain l_domain VARCHAR(191) DEFAULT 'unused' NOT NULL, CHANGE r_domain r_domain VARCHAR(191) DEFAULT '' NOT NULL;
+ALTER TABLE kam_users_missed_calls CHANGE from_domain from_domain VARCHAR(191) DEFAULT NULL, CHANGE ruri_domain ruri_domain VARCHAR(191) DEFAULT NULL;
+ALTER TABLE PeerServers CHANGE from_domain from_domain VARCHAR(191) DEFAULT NULL;
+ALTER TABLE CompanyAdmins CHANGE username username VARCHAR(64) NOT NULL;
+ALTER TABLE kam_users_acc CHANGE from_domain from_domain VARCHAR(191) DEFAULT NULL, CHANGE ruri_domain ruri_domain VARCHAR(191) DEFAULT NULL;
+ALTER TABLE kam_users_domain_attrs CHANGE did did VARCHAR(255) DEFAULT NULL;
+ALTER TABLE Users CHANGE username username VARCHAR(64) DEFAULT NULL, CHANGE maxCalls maxCalls SMALLINT DEFAULT 2 NOT NULL;
+ALTER TABLE BrandOperators CHANGE username username VARCHAR(64) NOT NULL;
+ALTER TABLE Friends CHANGE from_domain from_domain VARCHAR(191) DEFAULT NULL;
```

Comment:
Equivalence is with Friends.name (64) and Friends.domain (191).

Friend.from_domain is like PeerServers.from_domain (and some more), they are not related to our domain (that is now limited to 191) but to external domains. I think that changing them to 191 is not bad though.